dashboard/app/controllers/courses_controller.rb

```diff
@@ -20,4 +29,21 @@ def show
 
     render 'show', locals: {course: course}
   end
+
+  def new
+    render 'new'
+  end
+
+  def create
+    course = Course.new(name: params.require(:course).require(:name))
+    if course.save
+      redirect_to action: :edit, course_name: course.name
+    else
+      render 'new', locals: {error_messages: course.errors.full_messages}
```

Comment:
A common way of doing this is to set `@course`, and have the form in the view be `form_for @course`. That way any params are saved between attempts, and errors are already on the @course object and don't need to be passed separately.